release-a-chart-and-bump-to-upstream.md

```diff
@@ -0,0 +1,329 @@
+
+### Release a New Version
+
+Cherry-pick all changes against the chart on `master` branch, add another commit to amend the `version` field &/ other fields on `Chart.yaml`.
+
+After the PR to `release` branch is merged, a new chart version is released automatically.
+
+e.g. https://github.com/harvester/charts/pull/230
+
+e.g. https://github.com/harvester/charts/releases/tag/harvester-cloud-provider-0.2.4
+
+The chart like `harvester-cloud-provider` and `harvester-csi-driver` also need to be bumped to upstream RKE2 and Rancher. The following sections describe the details.
+
+:::note
+
+Always create/connect a github issue to the new chart release, QA and project management will cover all the requried processes.
+
+:::
+
+## Bump the Chart to `RKE2`
+
+### Update `rancher/rke2-charts`
+
+Point to the new release on Harvester chart repo.
+
+https://github.com/rancher/rke2-charts/blob/main-source/packages/harvester-cloud-provider/package.yaml
+
+e.g. https://github.com/rancher/rke2-charts/pull/454
+
+### Update `rancher/rke2` 
+
+Point to the new chart version, then `RKE2` will use the new chart version as the default value.
+
+https://github.com/rancher/rke2/blob/master/charts/chart_versions.yaml
+
+e.g. https://github.com/rancher/rke2/pull/5980
+
+:::note
+
+  Add PR to each active `rancher/rke2` branch.
+
+:::
+
+## Bump the Chart to `Rancher`
+
+### Bump the chart to `rancher/charts`
+
+Rancher charts are managed in a more complex way. First, read the document on this repo like https://github.com/rancher/charts/blob/dev-v2.9/README.md, https://github.com/rancher/charts/blob/dev-v2.9/docs/developing.md.
+
+We need the help of those two paths to step-by-step generate the target files.
+
+```
+  packages/harvester/harvester-cloud-provider/charts/  is used to download chart file in `make prepare`
+
+  packages/harvester/harvester-cloud-provider/generated-changes/ is used to put the generated patch files
+```
+
+Example: https://github.com/rancher/charts/pull/3234
+
+:::note
+
+  Add PR to each active `rancher/charts` branch.
+
+:::
+
+#### Update the package.yaml file
+
+Update the `url` and `version` fields.
+
+```
+diff --git a/packages/harvester/harvester-cloud-provider/package.yaml b/packages/harvester/harvester-cloud-provider/package.yaml
+index 7e86c3d2..5485f225 100644
+--- a/packages/harvester/harvester-cloud-provider/package.yaml
++++ b/packages/harvester/harvester-cloud-provider/package.yaml
+@@ -1,3 +1,3 @@
+-url: https://github.com/harvester/charts/releases/download/harvester-cloud-provider-0.1.14/harvester-cloud-provider-0.1.14.tgz
++url: https://github.com/harvester/charts/releases/download/harvester-cloud-provider-0.2.2/harvester-cloud-provider-0.2.2.tgz
+ version: 103.0.0
+ doNotRelease: false
+```
+
+:::note
+
+The Rancher version `103.0.0` needs to increase to a next version like `103.0.1` if `103.0.0` has been used, otherwise, the `index.yaml` will not be updated.
```

Comment:
could we refer to their readme section (https://github.com/rancher/charts?tab=readme-ov-file#upstream-charts/) for more detailed info?